pkg/controllers/configobservation/configobservercontroller/apply_configuration_test.go

```diff
@@ -0,0 +1,171 @@
+package configobservercontroller
+
+import (
+	"context"
+	"embed"
+	"fmt"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	operatorv1 "github.com/openshift/api/operator/v1"
+	configclient "github.com/openshift/client-go/config/clientset/versioned"
+	configinformer "github.com/openshift/client-go/config/informers/externalversions"
+	applyoperatorv1 "github.com/openshift/client-go/operator/applyconfigurations/operator/v1"
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/manifestclient"
+	libgoetcd "github.com/openshift/library-go/pkg/operator/configobserver/etcd"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/genericoperatorclient"
+	"github.com/openshift/library-go/pkg/operator/resourcesynccontroller"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	clocktesting "k8s.io/utils/clock/testing"
+)
+
+//go:embed testdata
+var packageTestData embed.FS
+
+func TestApplyConfiguration(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	mutationTrackingHTTPClient := manifestclient.NewHTTPClient("/home/deads/Downloads/must-gather-01")
+
+	kubeClient, err := kubernetes.NewForConfigAndClient(&rest.Config{}, mutationTrackingHTTPClient.GetHTTPClient())
+	if err != nil {
+		t.Fatal(err)
+	}
+	configClient, err := configclient.NewForConfigAndClient(&rest.Config{}, mutationTrackingHTTPClient.GetHTTPClient())
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	now, err := time.Parse(time.RFC3339, "2006-01-02T15:04:05Z")
+	if err != nil {
+		t.Fatal(err)
+	}
+	v1helpers.Clock = clocktesting.NewFakeClock(now)
```

Comment:
We're going to need an authoritative FakeClock to set for operators dealing with time unless we want to pass around fake clocks everywhere.  Thoughts?

Without this, conditions always come back with different transition times and that transition time is actually important to be able to predict.  Thinking this through, we'll need to have different times for each different test.  That suggests to me that we will *will* need to plumb the fake.Clock through per instance. Counter proposals?